Intermittent failure: websocktunnel: TestExponentialBackoffFailure

Intermittent failure in `tools/websocktunnel/client/client_test.go`:

```
=== RUN   TestExponentialBackoffFailure
time="2024-02-22T09:55:51Z" level=info msg="trying to connect to ws://127.0.0.1:42703"
time="2024-02-22T09:55:51Z" level=info msg="connection to ws://127.0.0.1:42703 failed -- retrying."
time="2024-02-22T09:55:51Z" level=info msg="trying to connect to ws://127.0.0.1:42703"
time="2024-02-22T09:55:51Z" level=info msg="trying to connect to ws://127.0.0.1:42703"
time="2024-02-22T09:55:51Z" level=info msg="connection to ws://127.0.0.1:42703 failed -- retrying."
time="2024-02-22T09:55:51Z" level=info msg="trying to connect to ws://127.0.0.1:42703"
time="2024-02-22T09:55:51Z" level=info msg="trying to connect to ws://127.0.0.1:42703"
time="2024-02-22T09:55:51Z" level=info msg="connection to ws://127.0.0.1:42703 failed -- retrying."
time="2024-02-22T09:55:52Z" level=info msg="trying to connect to ws://127.0.0.1:42703"
time="2024-02-22T09:55:52Z" level=info msg="trying to connect to ws://127.0.0.1:42703"
time="2024-02-22T09:55:52Z" level=info msg="connection to ws://127.0.0.1:42703 failed -- retrying."
time="2024-02-22T09:55:52Z" level=info msg="trying to connect to ws://127.0.0.1:42703"
time="2024-02-22T09:55:52Z" level=info msg="trying to connect to ws://127.0.0.1:42703"
time="2024-02-22T09:55:52Z" level=info msg="connection to ws://127.0.0.1:42703 failed -- retrying."
time="2024-02-22T09:55:52Z" level=info msg="trying to connect to ws://127.0.0.1:42703"
time="2024-02-22T09:55:52Z" level=info msg="trying to connect to ws://127.0.0.1:42703"
time="2024-02-22T09:55:52Z" level=info msg="connection to ws://127.0.0.1:42703 failed -- retrying."
time="2024-02-22T09:55:53Z" level=info msg="trying to connect to ws://127.0.0.1:42703"
time="2024-02-22T09:55:53Z" level=info msg="trying to connect to ws://127.0.0.1:42703"
time="2024-02-22T09:55:53Z" level=info msg="connection to ws://127.0.0.1:42703 failed -- retrying."
    client_test.go:124: wrong number of retries: 13
--- FAIL: TestExponentialBackoffFailure (2.00s)
```

See https://community-tc.services.mozilla.com/tasks/HddidoTjQvyECwXUFQJbpw/runs/0/logs/live/public/logs/live.log#L546

--- a/changelog/issue-6852.md
+++ b/changelog/issue-6852.md
@@ -1,0 +1,4 @@
+audience: general
+level: silent
+reference: issue 6852
+---

--- a/tools/websocktunnel/client/client_test.go
+++ b/tools/websocktunnel/client/client_test.go
@@ -114,13 +114,16 @@ func TestExponentialBackoffFailure(t *testing.T) {
 		t.Fatalf("should not run for more than %d milliseconds", maxTime)
 	}
 
-	// Note, sometimes 11 requests occur, presumably because the attempts occur
-	// when clock is 0ms, 200ms, 400ms, 600ms, 800ms, 1000ms, 1200ms, 1400ms,
-	// 1600ms, 1800ms, 2000ms, or there is a timeout go routine that isn't
-	// guaranteed to be immediately scheduled after 2 seconds. We may even
-	// want to bump the limit to 12, no need to be super strict here. Let's see
-	// how it is with 11 for now.
-	if count > 11 || count < 4 {
+	// Under load we have seen as many as 13 attempts before the timeout has
+	// taken effect, even though technically 13 attempts, which would includes
+	// 12x 200ms timeouts, should not be possible in 2s. But no need to be
+	// super strict here; websocktunnel doesn't run on a RTOS. Let's generously
+	// allow 15 retries.
+	//
+	// See:
+	//   * https://github.com/taskcluster/taskcluster/issues/6414
+	//   * https://github.com/taskcluster/taskcluster/issues/6852
+	if count > 15 || count < 4 {
 		t.Fatalf("wrong number of retries: %d", count)
 	}
 }

